graphql_compiler/cost_estimation/int_value_conversion.py

```diff
@@ -0,0 +1,94 @@
+    ))
+
+
+def convert_int_to_field_value(schema_info, vertex_class, property_field, int_value):
+    """Return the given integer's corresponding property field value.
+    Note that the property field values need not be integers. For example, all UUID values can be
+    converted to integers and vice versa, but they are provided in a string format (e.g.
+    00000000-0000-0000-0000-000000000000).
+    Example: If int_value is 9, and the property field is a uuid, then the resulting field
+    value will be 00000000-0000-0000-0000-000000000009.
+    Args:
+        schema_graph: SchemaGraph instance.
+        vertex_class: str, name of vertex class to which the property field belongs.
+        property_field: str, name of property field for which the domain of values is computed.
+        int_value: int, integer value which will be represented as a property field value.
+    Returns:
+        Any, the given integer's corresponding property field value.
+    Raises:
+        ValueError, if the given int_value is outside the range of valid values for the given
+        property field.
+    """
+    if is_int_field_type(schema_info, vertex_class, property_field):
+        return int_value
+    if is_datetime_field_type(schema_info, vertex_class, property_field):
+        return datetime.datetime.utcfromtimestamp(int_value)
+    if is_date_field_type(schema_info, vertex_class, property_field):
+        # TODO(bojanserafimov): this is not tested
+        return datetime.date.fromordinal(int_value)
+    elif is_uuid4_type(schema_info, vertex_class, property_field):
+        if not MIN_UUID_INT <= int_value <= MAX_UUID_INT:
+            raise AssertionError(u'Integer value {} could not be converted to UUID, as it'
+                                 u' is not in the range of valid UUIDs {} - {}: {} {}'
+                                 .format(int_value, MIN_UUID_INT, MAX_UUID_INT, vertex_class,
+                                         property_field))
+
+        return str(UUID(int=int(int_value)))
+    elif field_supports_range_reasoning(schema_info, vertex_class, property_field):
+        raise AssertionError(u'Could not represent int {} as {} {}, but should be able to.'
+                             .format(int_value, vertex_class, property_field))
+    else:
+        raise AssertionError(u'Could not represent int {} as {} {}. Currently,'
+                             u' only uniform uuid4 and int fields are supported.'
+                             .format(int_value, vertex_class, property_field))
+
+
+def convert_field_value_to_int(schema_info, vertex_class, property_field, value):
+    """Return the integer representation of a property field value."""
+    if is_int_field_type(schema_info, vertex_class, property_field):
+        return value
+    if is_datetime_field_type(schema_info, vertex_class, property_field):
+        return int((value - datetime.datetime(1970, 1, 1)).total_seconds())
```

Comment:
We are losing milisecond precision here for datetimes. How about running?
```
delta = (value - datetime.datetime(1970, 1, 1) 
return ((delta.days * 86400 + delta.seconds) * 10*6 + delta.microseconds) * 10**6
```
The code above is mostly copy pasted from the the `total_seconds` method of `timedelta` objects.
```
return ((self.days * 86400 + self.seconds) * 10**6 +
                self.microseconds) / 10**6
```

Reply:
Also, maybe should also use the UTC timezone for `datetime.datetime(1970, 1, 1)`?

Reply:
For the precision issue, just follow the [documentation's advice](https://docs.python.org/3/library/datetime.html#datetime.timedelta.total_seconds): `td / timedelta(microseconds=1)`

For the timezones, be mindful of the fact that datetime arithmetic requires that both operands are of the same flavor: timezone-aware or timezone-naive. Mixing them will cause errors.